proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,86 @@
+
+## Summary
+
+This proposal introduces a concept of a logical grouping for packages published to NuGet. This will allow one to easily identify all the packages released as part of a single product version release as well as request bulk-processing for a set of packages which belong to the same release. For example, all packages published as part of .NET 8 Preview 5 release will be staged together and then also can be deprecated together by a single call to NuGet APIs.
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many products (Apps / Tools / Frameworks / Utilities / etc. ...) consist of multiple interdependent packages which should be published together as a single set. Currently, NuGet provides no functionality for definding / grouping packages into such sets. This results in package owners having to keep track of these package sets on their own. There are three specific sets of scenarios which this feature will simplify dramatically:
+1. When preparing a release, product owners would like to have a mechanism to pre-stage their product packages, so that the release will consist only of a single action of marking the set of packages as `publishged`, which would result all the packages in the set to be marked as published at the same time.
+2. Sometimes developers have to take a dependency on several related packages in their projects, which are part of the same "release". Without having a concept of the `release` developers have to guess which versions of those packages are compatible. As a result, most of hte time developers would simply pick the latest version of all these packages, which somtimes can lead to incompatibility issues down the road.
+3. Keeping a package healthy on NuGet also means keeping the list of packages aligned with it support policy. Hence, package owners try to deprecate packages which are out of support. Without ability to group packages, package owners end up keeping track of all the packages which belong to the same release and run the deprecation over a discrete set of packages when necessary. Being able to group packages together would enable deprecating a whole release at once, knowing that all the packages which belong to that release will get deprecated.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+There is a new field in the package metadata, which indicates the unique release of the product the package belongs to. This field is called `release` and the value is scoped to the publisher of the package.
+
+When a package is being built, that field in the metadata is set to the same value for all the related packages, that are built together.
```

Comment:
Assuming that the .NET team end up using a release named ".NET 8.0.1", and later ".NET 8.0.2", what happens if some random person, not in the Microsoft or dotnet orgs, pushes a package using one of these release names?  What happens if someone not from Microsoft publishes a package using the ".NET 8.0.3" release name before Microsoft/dotnet does?

Reply:
That's exactly why I mentioned [here]() that the release should be scoped to the publisher. So that anyone else from another organization can have a release named exactly the same way and those will not interfere with each other by any means.